bat.d/41-framework

```diff
@@ -0,0 +1,536 @@
+            fi
+        fi
+    done # for step
+
+    echo_debug "bat" "batdrv.${_batdrv_plugin}.write_thresholds($1, $2, $3, $4).complete: bat=$_bat_str; rc=$rc"
+    return $rc
+}
+
+batdrv_chargeonce () {
+    # function not implemented
+    echo_debug "bat" "batdrv.${_batdrv_plugin}.charge_once.not_implemented"
+    return 255
+}
+
+batdrv_apply_configured_thresholds () {
+    # apply configured stop thresholds from configuration to all batteries
+    # output parameter errors only
+
+    if batdrv_select_battery BAT0; then
+        batdrv_write_thresholds 0 "$STOP_CHARGE_THRESH_BAT0" 1 "BAT0"; rc=$?
+    fi
+    if batdrv_select_battery BAT1; then
+        batdrv_write_thresholds 0 "$STOP_CHARGE_THRESH_BAT1" 1 "BAT1"; rc=$?
+    fi
+
+    return 0
+}
+
+batdrv_read_force_discharge () {
+    # function not implemented
+    echo_debug "bat" "batdrv.${_batdrv_plugin}.read_force_discharge.not_implemented"
+    return 255
+}
+
+batdrv_write_force_discharge () {
+    # function not implemented
+    echo_debug "bat" "batdrv.${_batdrv_plugin}.write_force_discharge.not_implemented"
+    return 255
+}
+
+batdrv_cancel_force_discharge () {
+    # function not implemented
+    echo_debug "bat" "batdrv.${_batdrv_plugin}.cancel_force_discharge.not_implemented"
+    return 255
+}
+
+batdrv_force_discharge_active () {
+    # function not implemented
+    echo_debug "bat" "batdrv.${_batdrv_plugin}.force_discharge_active.not_implemented"
+    return 255
+}
+
+batdrv_discharge () {
+    # function not implemented
+
+    # Important: release lock from caller
+    unlock_tlp tlp_discharge
+
+    echo_debug "bat" "batdrv.${_batdrv_plugin}.discharge.not_implemented"
+    return 255
+}
+
+batdrv_show_battery_data () {
+    # output battery status
+    # $1: 1=verbose
+    # global param: $_batteries
+    # prerequisite: batdrv_init()
+    local verbose=${1:-0}
+
+    printf "+++ Battery Care\n"
+    printf "Plugin: %s\n" "$_batdrv_plugin"
+
+    if [ "$_bm_thresh" != "none" ]; then
+        printf "Supported features: charge thresholds\n"
+    else
+        printf "Supported features: none available\n"
+    fi
+
+    printf "Driver usage:\n"
+    # native kernel ACPI battery API
+    case $_natacpi in
+        0)   printf "* natacpi (%s) = active (charge thresholds)\n" "$_batdrv_kmod" ;;
+        32)  printf "* natacpi (%s) = inactive (disabled by configuration)\n" "$_batdrv_kmod" ;;
+        128) printf "* natacpi (%s) = inactive (no kernel support)\n" "$_batdrv_kmod" ;;
+        254) printf "* natacpi (%s) = inactive (laptop not supported)\n" "$_batdrv_kmod" ;;
+        *)   printf "* natacpi (%s) = unknown status\n" "$_batdrv_kmod" ;;
+    esac
+    # TEMPLATE: customize to vendor specs
+    if [ "$_bm_thresh" != "none" ]; then
+        printf "Parameter value ranges:\n"
+		printf "* START_CHARGE_THRESH_BAT1/1:  0 (not available)\n"
+        printf "* STOP_CHARGE_THRESH_BAT1/1:   0..100(default)\n"
```

Comment:
Are you implying that the battery in Framework laptops is always called BAT1?

Reply:
My Framework 13 has `/sys/class/power_supply/BAT1`.
I don't know if this is specific to the Framework 13 though. Maybe the new Framework 16 is different - maybe @chutz can confirm :smile: